[#270] Standardise some Extra API (#360)
kowainik/relude: CHANGELOG.md

  Reimplement `ordNub` through `nubOrd` from `containers`.

  Add `intNub` and `intNubOn` functions.
* [#270](https://github.com/kowainik/relude/issues/270):
  Standardise `universe`, `universeNonEmpty` and `inverseMap` functions that
  previously were introduced in the `Relude.Extra.Enum` module. `Relude.Enum`
  module created that is exported in the main `Relude` module by default.

  __Migration guide:__ If you were using any of these functions you can now

>   remove `Relude.Extra.Enum` from your imports and they will be available for
>   you with the `Relude` module.

Reviewer: Not needed, but helps to cleanup the code
```suggestion
  remove `Relude.Extra.Enum` from your imports and explicit `mixins` section 
  as they are available for you with the `Relude` module.
```
